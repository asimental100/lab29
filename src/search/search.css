.mainDiv {
  display: block;
  margin: auto;
  background-color: cornsilk;
  margin-top: -20px;
  padding: 10px;
  margin: 0px;
}

.methodsList {
  list-style: none;
  margin-top: -15px;
  display: flex;
  flex-direction: column;
  margin-left: -25px;
}

button {
  display: block;
  margin: auto;
  margin-top: 10px;
  background-color: cyan;
  font-weight: 600;
}