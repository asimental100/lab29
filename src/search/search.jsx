import React from 'react';
import propTypes from 'prop-types';
import styles from './search.css'

export const Search = ({ onChange, onClick }) => (
    <div className={styles.mainDiv}>
      <div className={styles.search}>
        <label htmlFor="url"><h1>Search</h1></label>
        <input type="text" name="url" onChange={onChange} placeholder="Search" />
      </div>
      <div>
        <label htmlFor="method"><h1>Methods</h1></label>
        <ul className={styles.methodsList}>
          <li>GET<input type="radio" name="method" value="GET" onChange={onChange} /></li>
          <li>POST<input type="radio" name="method" value="POST" onChange={onChange} /></li>
          <li>PUT<input type="radio" name="method" value="PUT" onChange={onChange} /></li>
          <li>PATCH<input type="radio" name="method" value="PATCH" onChange={onChange} /></li>
          <li>DELETE<input type="radio" name="method" value="DELETE" onChange={onChange} /></li>
        </ul>
      </div>
      <div className={styles.json}>
        <label htmlFor="json"><h3>JSON (POST/PUT)</h3></label>
        <input type="text" name="json" placeholder="JSON" onChange={onChange} />
      </div>
      <button onClick={onClick}>Send!</button>
    </div>
);

Search.propType = {
  onChange: propTypes.func.isRequired,
  onClick: propTypes.func.isRequired
};
