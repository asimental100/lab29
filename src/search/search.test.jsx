import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { Search } from './search';

describe('Search component', () => {
  afterEach(() => cleanup());

  it('should test search component', () => {
    const { asFragment } = render(<Search 
      url="https://xfiles-api.herokuapp.com/api/v1/characters/Flukeman"
      method="GET"
      json=""
      onChange={() => {}}
      onClick={() => {}}
    />);
  expect(asFragment()).toMatchSnapshot();
  });
});
