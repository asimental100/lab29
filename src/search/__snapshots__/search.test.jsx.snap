// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component should test search component 1`] = `
<DocumentFragment>
  <div
    class="mainDiv"
  >
    <div
      class="search"
    >
      <label
        for="url"
      >
        <h1>
          Search
        </h1>
      </label>
      <input
        name="url"
        placeholder="Search"
        type="text"
      />
    </div>
    <div>
      <label
        for="method"
      >
        <h1>
          Methods
        </h1>
      </label>
      <ul
        class="methodsList"
      >
        <li>
          GET
          <input
            name="method"
            type="radio"
            value="GET"
          />
        </li>
        <li>
          POST
          <input
            name="method"
            type="radio"
            value="POST"
          />
        </li>
        <li>
          PUT
          <input
            name="method"
            type="radio"
            value="PUT"
          />
        </li>
        <li>
          PATCH
          <input
            name="method"
            type="radio"
            value="PATCH"
          />
        </li>
        <li>
          DELETE
          <input
            name="method"
            type="radio"
            value="DELETE"
          />
        </li>
      </ul>
    </div>
    <div
      class="json"
    >
      <label
        for="json"
      >
        <h3>
          JSON (POST/PUT)
        </h3>
      </label>
      <input
        name="json"
        placeholder="JSON"
        type="text"
      />
    </div>
    <button>
      Send!
    </button>
  </div>
</DocumentFragment>
`;
